# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaParent.h
# Commit: 46a5babaafd2
# Full Hash: 46a5babaafd2c0f2198f353d62254cf518dab391
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-15 04:16:11
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Implement initialization of entire persistent storage; r=dom-storage-reviewers,jari
#   
#   IndexedDB still calls QuotaManager::EnsurePersistentOriginIsInitializedInternal
#   from its idle database maintenance. However, we want to get rid of that so the
#   method can be made private. The idle database maintenance should only rely on
# ==============================================================================

diff -r 2ff5f3aaf841 -r 46a5babaafd2 dom/quota/QuotaParent.h
--- a/dom/quota/QuotaParent.h	Mon Oct 14 10:00:26 2024 +0000
+++ b/dom/quota/QuotaParent.h	Mon Oct 14 10:06:14 2024 +0000
@@ -43,6 +43,9 @@
   virtual mozilla::ipc::IPCResult RecvStorageInitialized(
       StorageInitializedResolver&& aResolver) override;
 
+  virtual mozilla::ipc::IPCResult RecvPersistentStorageInitialized(
+      PersistentStorageInitializedResolver&& aResolver) override;
+
   virtual mozilla::ipc::IPCResult RecvTemporaryStorageInitialized(
       TemporaryStorageInitializedResolver&& aResolver) override;
 
@@ -58,6 +61,9 @@
   virtual mozilla::ipc::IPCResult RecvInitializeStorage(
       InitializeStorageResolver&& aResolver) override;
 
+  virtual mozilla::ipc::IPCResult RecvInitializePersistentStorage(
+      InitializePersistentStorageResolver&& aResolver) override;
+
   virtual mozilla::ipc::IPCResult RecvInitializePersistentOrigin(
       const PrincipalInfo& aPrincipalInfo,
       InitializePersistentOriginResolver&& aResolve) override;